test/hotspot/jtreg/compiler/lib/template_framework/Template.java

```diff
@@ -0,0 +1,692 @@
+        }
+
+        /**
+         * Renders the {@link Template} to String.
+         *
+         * @param a The value for the first argument.
+         * @param b The value for the second argument.
+         * @return The String, resulting from rendering the {@link Template}.
+         */
+        public String render(A a, B b) {
+            return new TemplateToken.TwoArgs(this, a, b).render();
+        }
+
+        /**
+         * Renders the {@link Template} to String.
+         *
+         * @param a The value for the first argument.
+         * @param b The value for the second argument.
+         * @param fuel The amount of fuel provided for recursive Template instantiations.
+         * @return The String, resulting from rendering the {@link Template}.
+         */
+        public String render(float fuel, A a, B b) {
+            return new TemplateToken.TwoArgs(this, a, b).render(fuel);
+        }
+    }
+
+    /**
+     * Interface for function with three arguments.
+     *
+     * @param <T> Type of the first argument.
+     * @param <U> Type of the second argument.
+     * @param <V> Type of the third argument.
+     * @param <R> Type of the return value.
+     */
+    @FunctionalInterface
+    public interface TriFunction<T, U, V, R> {
+
+        /**
+         * Function definition for the three argument functions.
+         *
+         * @param t The first argument.
+         * @param u The second argument.
+         * @param v The third argument.
+         * @return Return value of the three argument function.
+         */
+        R apply(T t, U u, V v);
+    }
+
+    /**
+     * A {@link Template} with three arguments.
+     *
+     * @param arg0Name The name of the first argument, used for hashtag replacements in the {@link Template}.
+     * @param arg1Name The name of the second argument, used for hashtag replacements in the {@link Template}.
+     * @param arg2Name The name of the third argument, used for hashtag replacements in the {@link Template}.
+     * @param <A> The type of the first argument.
+     * @param <B> The type of the second argument.
+     * @param <C> The type of the third argument.
+     * @param function The function with three arguments that creates the {@link TemplateBody} given the template arguments.
+     */
+    public record ThreeArgs<A, B, C>(String arg0Name, String arg1Name, String arg2Name, TriFunction<A, B, C, TemplateBody> function) implements Template {
+        TemplateBody instantiate(A a, B b, C c) {
+            return function.apply(a, b, c);
+        }
+
+        /**
+         * Creates a {@link TemplateToken} which can be used as a {@link Token} inside
+         * a {@link Template} for nested code generation.
+         *
+         * @param a The value for the first argument.
+         * @param b The value for the second argument.
+         * @param c The value for the third argument.
+         * @return The template all (three) arguments applied.
```

Comment:
```suggestion
         * @return The template with all (three) arguments applied.
```

Reply:
Changed the wording completely, because we now return a `TemplateToken` :)